idb/postgres/internal/util/util.go

```diff
@@ -29,6 +29,7 @@ func TxWithRetry(db *pgxpool.Pool, opts pgx.TxOptions, f func(pgx.Tx) error, log
 		}
 
 		err = f(tx)
+		tx.Rollback(context.Background())
```

Comment:
if `tx.Commit` is called, then `tx.Rollback` wouldn't do anything; so logically it's ok. But it also seems to be wasteful to call `tx.Rollback` if it's not needed.

Reply:
agree with Will

Reply:
I think that the patten where `f()` is expected to call `Commit` is risky. I would rather seeing the `BeginTx`/`Commit`/`Rollback` all at the same place.

But that shouldn't be a blocker for this PR.

Reply:
I think the pattern we use in go-algorand is intuitive and safe -- `Atomic` manages all of this and a bit more. Maybe we can keep the code as-is and have a followup that does something like that.

Reply:
Created an issue to capture this idea: https://github.com/algorand/indexer/issues/688